server/src/controllers/user.controller.ts

```diff
@@ -43,6 +44,22 @@ export class UserController {
     return this.service.search(auth);
   }
 
+  @Get('me/activity')
+  @Authenticated({ permission: Permission.UserRead })
+  async getMyUploadActivity(
+    @Auth() auth: AuthDto,
+    @Query('from') from?: string,
+    @Query('to') to?: string,
+  ) {
+    // Default: last 52 weeks (end exclusive)
+    const end = to ? new Date(to) : new Date();
```

Comment:
there should't be any logic here, it should follow

Controller > Service > Repository layer